### PR TITLE
Add relationships

### PR DESCRIPTION
**Pull request by s-taylor**
This PR adds the ability to generate relationships when creating controllers. It will...
* Add a UUID property to the schema for the relationship
* Add a link to the schema for the relationship, so the related model can be looked up
* Add a test to the controller, to ensure the related model exists (this is achieved by a GET request to read the related model). It's a safeguard to ensure the user remembers to create the related model.

I needed pluralize to make this happen, so this builds on top of PR https://github.com/Prismatik/redbeard/pull/9

### CODE REVIEW

**Comment by davidbanham** [comment]
`redbeard controller argh -r foo`

Gives me:

```json
{
  "type": "object",
  "name": "{{name}}",
  "pluralName": "{{pluralName}}",
  "properties": {
    "id": {
      "type": "string",
      "faker": "random.uuid"
    },
    "name": {
      "type": "string",
      "faker": "name.findName"
    }
  },
  "": {
    "type": "string",
    "faker": "random.uuid",
    "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"
  },
  "links": [
    {
      "rel": "",
      "href": "//{}"
    }
  ],
  "required": ["name"]
}
```

Name and pluralName haven't been mustached and the links object isn't properly populated.

Am I doing it wrong?

**Comment by s-taylor** [comment]
That should work... let me investigate...

**Comment by s-taylor** [comment]
OK i've fixed the broken schema issue now

**Comment by davidbanham** [comment]
Wanna do the merge conflicts while you're at it?

**Comment by s-taylor** [comment]
Note: this only actually introduces singular (one-to-one) relationships. One-to-many should not be too difficult but requires thought on how we want to define the syntax when running the controller generator command.

**Comment by davidbanham** [comment]
I'm still getting name and pluralName not populating:

```json
{
  "type": "object",
  "name": "{{name}}",
  "pluralName": "{{pluralName}}",
  "properties": {
    "id": {
      "type": "string",
      "faker": "random.uuid"
    },
    "name": {
      "type": "string",
      "faker": "name.findName"
    }
  },
  "arghs": {
    "type": "string",
    "faker": "random.uuid",
    "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"
  },
  "links": [
    {
      "rel": "arghs",
      "href": "/arghs/{argh}"
    }
  ],
  "required": ["name"]
}
```

**Comment by s-taylor** [comment]
Merge conflicts should now be fixed.

**Comment by s-taylor** [comment]
@davidbanham I think this is finally fixed now... forgot to update the tags to `<%`

File: controller/schema.mustache [hunk 1]
```diff
@@ -12,5 +13,22 @@
       "faker": "name.findName"
     }
   },
+  <%#relationships%>
+  "<%pluralName%>": {
```
Comment (davidbanham) [inline controller/schema.mustache]: This should be singular rather than plural.

Comment (s-taylor) [inline controller/schema.mustache]: Fixed

Comment (davidbanham) [inline controller/schema.mustache]: Sorry, this was a total driveby comment by me. The relevant part of the spec is here:

http://json-schema.org/latest/json-schema-hypermedia.html#anchor3

That should give you some context on why it's the singular rather than the plural rather than just "That aint right".

**Comment by davidbanham** [comment]
LGTM!
